### PR TITLE
feat(java17): use JRE17, run tests on JRE17, compile with JDK11

### PR DESCRIPTION
**Pull request by mattgogerly**
Number 5 on our Java 17 adventure.

See also https://github.com/spinnaker/front50/pull/1274, https://github.com/spinnaker/igor/pull/1143 etc

### CODE REVIEW

**Comment by mattgogerly** [comment]
@Mergifyio update

**Comment by mattgogerly** [comment]
@Mergifyio update

File: gradle.properties [hunk 1]
```diff
@@ -2,6 +2,7 @@ fiatVersion=1.42.0
 korkVersion=7.204.0
 kotlinVersion=1.5.32
 org.gradle.parallel=true
+org.gradle.jvmargs=-Xmx6144m
```
Comment (dbyron-sf) [inline gradle.properties:5]: nit: I'd use what we're telling [github actions to use](https://github.com/spinnaker/orca/blob/7a95d5a7446329dd2fa99ba67019b5077c7760ac/.github/workflows/build.yml#L10)
```
-Xmx6g -Xms6g
```

**Review by dbyron-sf — APPROVED** [review]
one nit, but approved.  Pretty sure [buildtool](https://github.com/spinnaker/buildtool/blob/8c1c18ee1e6e2775e9db1eafb22ef59a44e039a1/dev/buildtool/__init__.py#L5) needs an update too.

**Pull request merged**